services/src/main/java/org/fao/geonet/api/selections/FavouriteMetadataListApi.java

```diff
@@ -0,0 +1,682 @@
+    FavouriteMetadataListVM getFavouritesList(
+        @Parameter(description = "FavouritesList DB id (int)",required = true)
+        @PathVariable
+        Integer favouriteListIdentifier,
+        @Parameter(hidden = true)
+        HttpSession httpSession,
+        @Parameter(hidden = true)
+        HttpServletRequest  httpServletRequest
+    )
+        throws Exception {
+
+        if (favouriteListIdentifier == null) {
+            throw new IllegalArgumentException("no favouriteListIdentifier given");
+        }
+        if (favouriteListIdentifier <=0) {
+            throw new IllegalArgumentException("invalid favouriteListIdentifier given");
+        }
+
+        boolean isAdmin = isAdmin(httpSession);
+        String sessionId = getSessionId(httpServletRequest);
+        User user =  getUser(httpSession);
+
+        Optional<FavouriteMetadataList> list = favouriteMetadataListRepository.findById(favouriteListIdentifier);
+        if (!list.isPresent()) {
+            throw new ResourceNotFoundException(String.format("cannot find favouriteListIdentifier, based on id %d", favouriteListIdentifier));
+        }
+        FavouriteMetadataList result = list.get();
+
+        if (permittedRead(result,user,sessionId,isAdmin)) {
+            return new FavouriteMetadataListVM(result);
+        }
+
+        throw new NotAllowedException("you don't have permission to read that favouriteList list");
+    }
+
+    /**
+     *  update a FavouriteMetadataList
+     *  user or session must own the FavouriteMetadataList.
+     *  change name - name must be unique for the user/session.
+     *
+     *  update the set of metadataUuuids (add/remove/replace)
+     *
+     */
+    @io.swagger.v3.oas.annotations.Operation(summary = "Update a specific (by DB id) of FavouriteList list")
+    @RequestMapping(
+        method = RequestMethod.PUT,
+        value = "/{favouriteListIdentifier}",
+        produces = {
+            MediaType.APPLICATION_JSON_VALUE
+        })
+    public
+    @ResponseBody
+    @ResponseStatus(HttpStatus.OK)
+    FavouriteMetadataListVM updateFavouriteList(
+        @Parameter(description = "favouriteList DB id (int)",required = true)
+        @PathVariable
+            Integer favouriteListIdentifier,
+
+        @Parameter(description = "new name of list",required = false)
+        @RequestParam(name="name",required = false)
+            String name,
+
+        @Parameter(description = "action - add, replace, or remove",required = true)
+        @RequestParam(name="action",required = true)
+            ActionType action,
+
+        @Parameter(description = "List of metadata uuids to replace in the list",required = false)
+        @RequestParam(name="metadataUuids",required = false)
+            String[] metadataUuids,
+
+        @Parameter(hidden = true)
+            HttpSession httpSession,
+        @Parameter(hidden = true)
+            HttpServletRequest  httpServletRequest
+    )
+        throws Exception {
```

Comment:
The sessionId cookie can be retrieved using the `@CookieValue` annotation:
```suggestion
    public
    @ResponseBody
    @ResponseStatus(HttpStatus.OK)
    FavouriteMetadataListVM updateFavouriteList(
        @CookieValue(SESSION_COOKIE_NAME) sessionId,
        @Parameter(description = "favouriteList DB id (int)",required = true)
        @PathVariable
            Integer favouriteListIdentifier,

        @Parameter(description = "new name of list",required = false)
        @RequestParam(name="name",required = false)
            String name,

        @Parameter(description = "action - add, replace, or remove",required = true)
        @RequestParam(name="action",required = true)
            ActionType action,

        @Parameter(description = "List of metadata uuids to replace in the list",required = false)
        @RequestParam(name="metadataUuids",required = false)
            String[] metadataUuids,

        @Parameter(hidden = true)
            HttpSession httpSession,
        @Parameter(hidden = true)
            HttpServletRequest  httpServletRequest
    )
        throws Exception {
```